bfdriver/src/common.c

```diff
@@ -425,9 +425,11 @@ common_start_vmm(void)
             break;
     }
 
+    platform_call_vmm_on_core(0, BF_REQUEST_VMM_INIT, (uint64_t)platform_num_cpus(), 0);
+
     for (cpuid = 0, g_num_cpus_started = 0; cpuid < platform_num_cpus(); cpuid++) {
         ret = platform_call_vmm_on_core(
-                  (uint64_t)cpuid, BF_REQUEST_VMM_INIT, (uint64_t)cpuid, 0);
+                  (uint64_t)cpuid, BF_REQUEST_VMM_RUN, (uint64_t)cpuid, 0);
```

Comment:
- object construction cannot occur on a single core as that will mess up the thread objects. Each vCPU needs to be created on the core it will run on to make sure that the thread objects are initialized properly. 
- same thing, we cannot create the vCPUs all at the same time. They have to be done one by one.
- A default domain can be created as a global object without issue. The domain's constructor is called during global construction on Core 0. 

In other words, no additional function is needed

Reply:
Can you point me to where thread object initialization happens?